crates/wdk-build/src/resource_compile.rs

```diff
@@ -0,0 +1,217 @@
+}
+
+// function to invoke RC.exe
+fn invoke_rc(includeargs: &Vec<String>, rc_exe_rootpath: String) {
+
+    let resource_script = "resources.rc";
+    let rc_exe_path = format!("{}\\rc.exe", rc_exe_rootpath);
+    let rc_exe_path = Path::new(&rc_exe_path);
+    if !rc_exe_path.exists() {
+        eprintln!(
+            "Error: rc.exe path does not exist : {}", 
+            rc_exe_path.display()
+        );
+        std::process::exit(1); // Exit with a non-zero status code
+    }
+
+    let mut command = Command::new(rc_exe_path);
+    command.args(includeargs).arg(resource_script);
+    println!("Command executed: {:?}", command);
+    
+    let status = command.status();
+
+    match status {
+        Ok(exit_status) => {
+            if exit_status.success() {
+                println!("Resource compilation successful!");
+                println!("cargo:rustc-link-arg=resources.res");
+            } else {
+                println!("Resource compilation failed.");
+                std::process::exit(1); // Exit with a non-zero status code
+            }
+        }
+        Err(err) => {
+            eprintln!("Error running rc.exe: {}", err);
+            std::process::exit(1); // Exit with a non-zero status code
+        }
+    }
+}
+// function to get package metadata details
+fn get_packagemetadatadetails() -> (String, String, String) {
+    // Run the 'cargo metadata' command and capture its output
+    let path = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let meta = MetadataCommand::new()
+        .manifest_path("./Cargo.toml")
+        .current_dir(&path)
+        .exec()
+        .unwrap();
+    let root = meta.root_package().unwrap();
+    let metadata = &root.metadata;
+
+    // Extract metadata values with default fallbacks
+    let companyname = metadata
+        .get("wdk")
+        .and_then(|wdk| wdk.get("driver-model"))
+        .and_then(|driver_model| driver_model.get("companyname"))
+        .map(|s| s.to_string())
+        .unwrap_or_else(|| "Company name not found in metadata".to_string());
+    let copyrightname = metadata
+        .get("wdk")
+        .and_then(|wdk| wdk.get("driver-model"))
+        .and_then(|driver_model| driver_model.get("copyright"))
+        .map(|s| s.to_string())
+        .unwrap_or_else(|| "Copyright name not found in metadata".to_string());
+    let productname = metadata
+        .get("wdk")
+        .and_then(|wdk| wdk.get("driver-model"))
+        .and_then(|driver_model| driver_model.get("productname"))
+        .map(|s| s.to_string())
+        .unwrap_or_else(|| "Product name not found in metadata".to_string());
+
+    (companyname, copyrightname, productname)
+}
+// function to get package details
+fn get_packagedetails() -> (String, String, String, String) {
+    let mut fileversion = String::new();
+    let mut description = String::new();
+    let mut productversion = String::new();
+    let mut name = String::new();
+
+    match fs::read_to_string("Cargo.toml") {
```

Comment:
This is very hacky to get the crate version. You should use the env vars that cargo provides you: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates